src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java

```diff
@@ -83,23 +90,67 @@ public void setDensity(float density)
     }
 
     /**
-     * Event that allows any feature to customize the rendering of fog.
+     * This event allows for customization of parameters related to fog rendering. The plane distances are based on the player's render distance.
+     *
+     * A FogMode of FOG_SKY is the skybox's fog.
+     * A FogMode of FOG_TERRAIN is what the fog induced by render distance uses. This works best for reducing the camera's visibility.
      */
-    @HasResult
-    public static class RenderFogEvent extends FogEvent
+    public static class RenderFogEvent extends EntityViewRenderEvent
     {
+        private final FogMode type;
         private final float farPlaneDistance;
+        private final float nearPlaneDistance;
+        private final FogShape fogShape;
 
+        @Deprecated //TODO: remove in 1.19
         public RenderFogEvent(FogMode type, Camera camera, float partialTicks, float distance)
         {
-            super(type, camera, partialTicks);
-            this.farPlaneDistance = distance;
+            this(type, camera, partialTicks, 0f, distance, FogShape.SPHERE);
+        }
+
+        public RenderFogEvent(FogMode type, Camera camera, float partialTicks, float nearPlaneDistance, float farPlaneDistance, FogShape fogShape)
+        {
+            super(Minecraft.getInstance().gameRenderer, camera, partialTicks);
+            this.type = type;
+            this.farPlaneDistance = farPlaneDistance;
+            this.nearPlaneDistance = nearPlaneDistance;
+            this.fogShape = fogShape;
+        }
+
+        public FogMode getMode()
+        {
+            return type;
         }
 
         public float getFarPlaneDistance()
         {
             return farPlaneDistance;
         }
+
+        public float getNearPlaneDistance()
+        {
+            return nearPlaneDistance;
+        }
+
+        public FogShape getFogShape()
+        {
+            return fogShape;
+        }
+
+        public void setFarPlaneDistance(float distance)
+        {
+            RenderSystem.setShaderFogEnd(distance);
+        }
+
+        public void setNearPlaneDistance(float distance)
+        {
+            RenderSystem.setShaderFogStart(distance);
+        }
+
+        public void setFogShape(FogShape shape)
+        {
+            RenderSystem.setShaderFogShape(shape);
+        }
```

Comment:
Currently, this sets the near and far planes and shape immediately, without changing the values returned by the getters. This may pose possible mod compatibility problems if multiple mods wish to simply "add on" to the near/far plane distances, as currently they would always overwrite any previously set distance.

Perhaps it may be better to store the originals, and have mutable fields in the event with getters and setters, which are then provided to `RenderSystem` in the hook?